### PR TITLE
Fix fieldmap nfft copy

### PR DESCRIPTION
**Pull request by alexjaffray**
Fix copying of `FieldmapNFFTOp` to return a copy that is the same size in memory as the original op. Prior to this PR, copying a FieldmapNFFTOp would return a copy larger than the original in memory.

### CODE REVIEW

File: MRIOperators/src/FieldmapNFFTOp.jl [hunk 1]
```diff
@@ -93,45 +97,48 @@ function FieldmapNFFTOp(shape::NTuple{D,Int64}, tr::Trajectory,
...
   x_tmp = zeros(Complex{T}, S.ncol)
   y_tmp = zeros(Complex{T}, S.nrow)
 
-  cparam = deepcopy(S.cparam)
+  cparam = copy(S.cparam)
+  d = [zeros(Complex{T}, length(idx[κ])) for κ=1:K]
+  p = [zeros(Complex{T}, shape) for κ=1:K]
 
-  mul! = (res,x) -> produ!(res,x,x_tmp,S.shape,plans,idx,cparam,S.circTraj,d,p)
-  ctmul! = (res,y) -> ctprodu!(res,y,y_tmp,S.shape,plans,idx,cparam,S.circTraj,d,p)
```
Comment (JeffFessler) [inline MRIOperators/src/FieldmapNFFTOp.jl:127]: personally i prefer having definitions like this separated like they were here
so that one can easily inspect them with @show etc.
rather than embedded in a long function call...
was this change needed to avoid the memory issue?

Comment (alexjaffray) [reply MRIOperators/src/FieldmapNFFTOp.jl:127]: @JeffFessler separating them seems to have no impact on the memory issue, I will separate them and call them mul! and ctmul! as before

Comment (alexjaffray) [reply MRIOperators/src/FieldmapNFFTOp.jl:127]: ^pushed the change

**Comment by tknopp** [comment]
looks good
